imap/message.go

```diff
@@ -505,6 +507,15 @@ func createMessage(c *protonmail.Client, u *protonmail.User, privateKeys openpgp
 				pw.CloseWithError(cleartext.Close())
 			}()
 
+			go func () {
+				b := bytes.Buffer{}
+				if _, err := io.Copy(&b, spr); err != nil {
+					pw.CloseWithError(err)
+					return
+				}
+				att.Signature = b.String()
```

Comment:
I think this is racy. There's no guarantee `att.Signature` will be set when the `createMessage` function returns.

But if everything we're doing is writing the signature to a bytes buffer, there's no need for an extra goroutine and pipe in the first place. We can just:

```go
var sig bytes.Buffer
att.Encrypt(pw, &sig, privateKey)
```

And then we can set `att.Signature` after `pr` has been completely consumed (ie. after `CreateAttachment`).